### PR TITLE
Desktop polish: Now Playing bar/view, panes, window state, density, hover, context menus, multi-select (#380-#387)

### LINKED ISSUE #380: [Desktop] Add a desktop Now Playing bar
Part of #376.

## Goal
Keep core playback controls visible while browsing on desktop with a persistent bottom Now Playing bar.

## Likely code areas
- `lib/features/player/mini_player.dart`
- `lib/features/player/player_providers.dart`
- shared playback controls/widgets
- desktop app shell/layout
- player widget tests

## Requirements
- Show artwork, title, artist, play/pause, previous/next, progress/seek and volume/mute.
- Clicking the bar should open the full Now Playing view.
- Use shared playback state and commands; do not create a desktop-only playback controller.
- Handle no-track, loading, buffering and playback-error states cleanly.
- Keep the existing mobile mini-player behavior unchanged.

## Acceptance criteria
- [ ] Bar remains visible while navigating desktop library views.
- [ ] Controls update immediately from shared playback state.
- [ ] Seeking and volume work from mouse/keyboard.
- [ ] Layout remains usable at the minimum supported desktop width.
- [ ] Tests cover no-track, playing, paused and loading states.

### LINKED ISSUE #381: [Desktop] Add a full desktop Now Playing view
Part of #376.

## Goal
Create a desktop-first Now Playing view that uses wide space intentionally instead of stretching the phone screen.

## Likely code areas
- `lib/features/player/player_screen.dart`
- `lib/features/player/widgets/`
- `lib/features/player/player_providers.dart`
- lyrics and queue widgets/tests

## Requirements
- Large artwork plus track/album/artist metadata.
- Full transport, seek, shuffle/repeat and volume controls.
- Lyrics and queue can sit side-by-side on wide windows.
- Reuse shared playback/lyrics/queue state; no Linux-only state model.
- Collapse cleanly for narrower windows and preserve mobile layout.

## Acceptance criteria
- [ ] Wide layout has clear artwork, controls, lyrics and queue hierarchy.
- [ ] Resizing does not lose state or overflow.
- [ ] Keyboard focus order is sensible.
- [ ] Existing Android Now Playing behavior is unchanged.
- [ ] Widget tests cover wide/narrow breakpoints.

### LINKED ISSUE #382: [Desktop] Add responsive two- and three-pane layouts
Part of #376.

## Goal
Use wide Linux windows with intentional two- and three-pane layouts instead of stretched mobile screens.

## Likely code areas
- desktop app shell/layout widgets
- `lib/features/library/`
- `lib/features/player/`
- responsive/shared widgets and tests

## Requirements
- Introduce shared desktop breakpoints/layout helpers.
- Support combinations such as navigation + content + detail/queue where useful.
- Preserve route/state identity while panes appear/disappear on resize.
- Avoid duplicating entire Android and Linux screen trees.

## Acceptance criteria
- [ ] Common desktop widths use space efficiently.
- [ ] Resizing between pane modes does not reset selected content/playback.
- [ ] No clipped/overflowing controls at minimum width.
- [ ] Mobile/narrow layout remains unchanged.
- [ ] Responsive behavior has widget-test coverage.

### LINKED ISSUE #383: [Desktop] Persist window size and position
Part of #376.

## Goal
Remember the Linux window size, maximized state and position where the desktop/window manager supports it.

## Likely code areas
- Linux window/runner integration under `linux/`
- Dart startup/app-shell state
- persisted preferences
- desktop startup tests

## Requirements
- Restore a sane last-known size and maximized state on launch.
- Persist only desktop window state; do not affect Android.
- Handle invalid/off-screen saved positions after monitor changes.
- Keep a safe first-launch default and minimum size.
- Avoid fighting the window manager on Wayland where exact position restoration is unavailable.

## Acceptance criteria
- [ ] Size/maximized state survive restart.
- [ ] Invalid saved geometry falls back safely.
- [ ] Multi-monitor/disconnected-monitor cases do not strand the window off-screen.
- [ ] Mobile behavior is unchanged.
- [ ] State parsing/persistence has focused tests.

### LINKED ISSUE #384: [Desktop] Add desktop-friendly content density
Part of #376.

## Goal
Tune desktop information density so normal Linux windows show more useful library content than the touch-first Android layout.

## Likely code areas
- `lib/features/library/`
- shared track/album/artist widgets
- desktop responsive/theme helpers
- widget/golden tests

## Requirements
- Reduce unnecessary row height/padding on desktop while keeping controls easy to click.
- Increase album/artist grid density based on available width.
- Keep typography, artwork proportions and Linthra visual identity consistent.
- Do not globally shrink mobile/touch targets.
- Coordinate with #395 so density preferences can build on a clean desktop baseline.

## Acceptance criteria
- [ ] Desktop lists/grids fit materially more content.
- [ ] Text/artwork do not clip at supported scaling factors.
- [ ] Touch/mobile density remains unchanged.
- [ ] Wide/narrow layouts have focused widget coverage.

### LINKED ISSUE #385: [Desktop] Add hover states to interactive controls
Part of #376.

## Goal
Add subtle, consistent hover feedback so mouse users can tell what is interactive.

## Likely code areas
- shared buttons/cards/list-row widgets
- `lib/features/library/`
- desktop sidebar/player controls
- widget tests

## Requirements
- Add hover states for album/artist cards, track rows, buttons, sidebar destinations and menus.
- Keep hover visual changes subtle and compatible with light/dark themes.
- Do not treat hover as selection, focus or pressed state.
- Touch/mobile behavior must remain unchanged.
- Prefer reusable desktop interaction helpers over repeated MouseRegion code.

## Acceptance criteria
- [ ] Main desktop interactive surfaces show visible hover feedback.
- [ ] Hover does not interfere with keyboard focus or selected states.
- [ ] Theme contrast remains acceptable.
- [ ] Tests cover at least representative card, row and navigation controls.

### LINKED ISSUE #386: [Desktop] Add right-click context menus
Part of #376.

## Goal
Provide desktop right-click context menus for library entities without changing mobile long-press behavior.

## Likely code areas
- `lib/features/library/widgets/track_tile.dart`
- album/artist/playlist widgets
- queue/download/favorite actions
- shared menu/action helpers and tests

## Requirements
- Track menu: Play next, Add to queue, Add to playlist, Favorite, Download/Remove download, Show album, Show artist where applicable.
- Album/artist/playlist menus should expose only actions already supported by the domain layer.
- Reuse existing commands/providers; context menus must not duplicate business logic.
- Keyboard users need an equivalent way to open/operate menus.
- Menu content must reflect current state (downloaded/favorited/etc.).

## Acceptance criteria
- [ ] Right-click opens the correct menu on supported desktop entities.
- [ ] Actions update shared state correctly.
- [ ] Disabled/unavailable actions are handled clearly.
- [ ] Mobile long-press behavior is unchanged.
- [ ] Representative context-menu actions have widget tests.

### LINKED ISSUE #387: [Desktop] Add multi-select for tracks
Part of #376.

## Goal
Support normal desktop multi-selection for track lists.

## Likely code areas
- `lib/features/library/widgets/track_tile.dart`
- track-list selection state/providers
- queue/playlist/download bulk actions
- widget tests

## Requirements
- Ctrl-click toggles individual tracks; Shift-click selects a contiguous range.
- Selection must survive harmless list rebuilds but clear predictably when changing context.
- Bulk actions should reuse existing queue/playlist/download commands.
- Current playing state and selected state must remain visually distinct.
- Keyboard selection should work where practical.

## Acceptance criteria
- [ ] Ctrl/Shift selection behaves predictably.
- [ ] Bulk queue/playlist/download actions receive exactly the selected tracks.
- [ ] Sorting/filtering cannot accidentally apply actions to the wrong rows.
- [ ] Mobile behavior remains unchanged.
- [ ] Selection edge cases have tests.

### PR DESCRIPTION
**Pull request by TheZupZup**
Eight issues from the desktop milestone (#376), one commit each so they can be read (or dropped) separately.

Closes #380
Closes #381
Closes #382
Closes #383
Closes #384
Closes #385
Closes #386
Closes #387

## What each commit does

**#380 - the now-playing bar is seekable.** The bar already carried the whole transport on a desktop window; the progress line was a readout, so jumping back thirty seconds meant opening the full player. On a desktop host it is now the same `PlaybackProgressBar` the full player uses, at a new compact density and without the time caption. Reusing it means the same drag preview, the same optimistic hold until playback acknowledges the seek, the same slider semantics and arrow keys.

**#381 - the queue sits beside the cover.** A window wide enough for a third column draws the queue as that column instead of a sheet over the top, so lyrics and up-next are readable at once. Same `QueueSheet`, in an embedded mode; the action row's queue button becomes the pane toggle. Narrower windows keep the sheet.

**#382 - the library gets a detail pane, and the pane layout is shared.** Album and artist detail had each hand-rolled the same desktop composition down to their own copy of the 320 px constant; that is now `SplitPanes` in `shared/layout`. And picking an album from the grid no longer replaces the whole view: past `listDetailMinWidth` the grid stays and the album opens beside it, which with the navigation rail is navigation · content · detail. Below that width the same screen is pushed as a route, as before.

**#383 - the window remembers its geometry.** Size and maximized state survive a restart; position too, on X11 (Wayland gives a client neither its own position nor a way to set one). The size saved is the *unmaximized* one, since GTK will not tell you that after the fact. A saved position is re-checked against the monitors attached now, so unplugging a monitor drops the position rather than restoring the window off-screen — the size survives either way. The rules live in a new `native/linthra_desktop`: pure C++, no GTK, no filesystem, unit-tested without a display server. `check_linux_runner.py` grows a guard, because losing any of the wiring still compiles and still launches.

**#384 - content density follows the pointer.** The theme uses `VisualDensity.adaptivePlatformDensity`: compact on desktop, standard everywhere touch-first. The two places that sized rows with a hard-coded number now follow it — the songs list's fixed row extent (which the A–Z index measures its scroll offsets in) and the artist grid's cell floor — and both only give back padding.

**#385 - hover you can actually see.** Material's 4% default is invisible on a black-first theme. One `hoverColor` reaches every ink surface at once, neutral rather than brand-tinted so it stays distinct from selection. `HoverHighlight` covers the one case ink cannot: an album cover hides the overlay painted on the Material behind it.

**#386 - right-click context menus.** `ContextMenuRegion` opens a surface's existing menu on secondary click and on the keyboard's menu key (or Shift+F10). Track rows share one list and one dispatcher with their 3-dot button, and gain Show album / Show artist. Albums and artists get Play · Shuffle · Play next · Add to queue · Add to playlist, each a command their detail pages already call. Playlists reuse their rename/delete pair.

**#387 - Ctrl and Shift multi-select.** Ctrl (or Cmd) picks one row out and starts a selection when there is none; Shift takes everything between. `TrackSelection` holds it for the songs list and album/artist detail, keyed by the provider-namespaced uri, and resolves a bulk action against the list as shown. Escape leaves the selection.

## Review fixes

Four review rounds landed on top of the eight, each with a test verified to fail without its fix. Grouped, they were:

- **Window state (#383):** rectangle arithmetic in `int64_t` (a coordinate near `INT_MAX` was signed-overflow UB, confirmed under UBSan), the saved position validated against the geometry the window actually opens with rather than the one the file claimed, the destroy handler disconnected with the other two, and `check_linux_runner.py` looking for the save inside `my_application_shutdown()` specifically.
- **Menus (#386):** the anchor converted into the overlay's coordinate space (every menu was sliding across by the navigation rail's width), Show album / Show artist gated on catalog membership so a folder-browsed track is not offered a page that says "not found", and a pick dropped when the region that opened it is gone.
- **Panes (#381, #382, #387):** the panes are widgets in a layout rather than routes, so a resize unmounts them mid-action. The queue's "save as playlist" captures its repository before the name prompt; the panes' track selection is held by `LibraryScreen` so a resize does not discard it; and leaving a selection no longer depends on the screen still being mounted.
- **Selection (#387):** `SelectionEscapeScope`, so Escape leaves a selection on the detail pages too, not just the songs list.

## Testing

- `flutter analyze` clean, `flutter test` green (4693 tests, ~140 new).
- New C++ tests for the window-state policy: `ctest --test-dir build/linthra_desktop`, wired into a new `cpp-desktop-window.yml` workflow alongside the audio DSP one.
- `scripts/check_linux_runner.py` and its tooling tests pass, including the new cases for the window-state wiring.
- `linux/runner/window_state_store.cc` and `my_application.cc` compile clean against GTK 3 with `-Wall -Werror`; the full Linux build is CI's to prove.

## Notes

- Two existing tests in `detail_desktop_layout_test.dart` now open the detail screens on their own route rather than by tapping the grid. They test the detail screen's internal two-pane layout, and on a wide Library that tap now opens a pane instead — the route is still how Folders, search and a narrow window reach it.
- Playlist detail keeps its own selection: its rows are a reorderable list rather than `TrackTile`, so bringing it onto `TrackSelection` is a separate change.
- Position restoration and the GTK half of #383 can only really be judged on a real session — worth a manual pass on both X11 and Wayland, and with a monitor unplugged between launches.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01YWjKwW5dZSDVXUeixZkf7e

### CODE REVIEW

File: native/linthra_desktop/src/window_state.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+// The overlap between two rectangles, in pixels. Zero on either axis means they
+// do not meet.
+void Intersection(const WindowGeometry& window, const WorkArea& area,
+                  int* overlap_width, int* overlap_height) {
+    const int left = std::max(window.x, area.x);
+    const int right = std::min(window.x + window.width, area.x + area.width);
+    const int top = std::max(window.y, area.y);
+    const int bottom = std::min(window.y + window.height, area.y + area.height);
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply native/linthra_desktop/src/window_state.cpp]: Confirmed and fixed in 5a2596c. The rectangle arithmetic is done in `int64_t` now, so every value the parser can produce adds and subtracts exactly and the rejection happens on the numbers.

Reproduced first: built the old code with `-fsanitize=undefined -fno-sanitize-recover=all` and it stops on `signed integer overflow: 1200 + 2147483647 cannot be represented in type 'int'`. The same run against the fixed source passes. The test that drives it goes through the parser, since that is where such a value actually comes from, and covers both ends of the range.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: native/linthra_desktop/src/window_state.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    } else {
+        // A corrupt size says nothing about whether the window was maximized,
+        // so a first-launch default is the honest answer for both.
+        resolved.width = limits.default_width;
+        resolved.height = limits.default_height;
+    }
+    resolved.width = std::max(resolved.width, limits.min_width);
+    resolved.height = std::max(resolved.height, limits.min_height);
+
+    if (IsPositionUsable(saved, work_areas)) {
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply native/linthra_desktop/src/window_state.cpp]: Right, and your example is exactly the failing case. Fixed in 5a2596c: the position is now checked against the geometry the window will actually open with, not the one the file claimed.

`width=1, height=1, x=1919, y=1049` on a 1920×1050 work area is a test now — it resolves to 420×600 with no position. Verified it fails without the fix.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: linux/runner/window_state_store.cc [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    // no saved position and still wants to remember the one it ends up with.
+    self->tracked.has_position = SessionHasWindowPositions(window);
+    self->maximized = resolved.maximized ? TRUE : FALSE;
+    self->fullscreen = FALSE;
+
+    self->configure_handler = g_signal_connect(
+        window, "configure-event", G_CALLBACK(OnConfigure), self);
+    self->state_handler = g_signal_connect(
+        window, "window-state-event", G_CALLBACK(OnWindowState), self);
+    g_signal_connect(window, "destroy", G_CALLBACK(OnDestroy), self);
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply linux/runner/window_state_store.cc]: Good catch. Fixed in 5a2596c: the destroy handler's id is kept and disconnected with the other two.

Freeing the store while its window is still alive is the unusual order, but it is reachable — a disposal, or a startup unwound before GTK destroys anything — and leaving that one handler registered would hand `OnDestroy` a freed struct the moment the window did go away.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/check_linux_runner.py [hunk 4]
```diff
@@ -926,6 +930,46 @@ def _function_body(code: str, signature: str, where: Path) -> tuple[int, int]:
...
+            f"{WINDOW_STATE_POLICY_SOURCE.name} policy needs"
+        )
+
+    my_application = _read(root, MY_APPLICATION)
+    if "window_state_store_new" not in my_application:
+        problems.append(
+            f"{MY_APPLICATION} never calls window_state_store_new(), so the "
+            "saved window geometry is never restored"
+        )
+    if "window_state_store_save" not in my_application:
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply scripts/check_linux_runner.py]: Correct — and it was the merge with #401 that introduced the second save, so the guard was weakened by exactly the change that added it. Fixed in 5a2596c: the checker parses `my_application_shutdown()`'s body with the same `_function_body` helper the identity checks use, and looks for the call there.

Two guardrail tests, both verified to fail against the old substring check: one drops only the shutdown save (leaving the one in `activate()` in place), and one comments it out — the blanked-comments pass is what stops a call that is merely *described* from standing in for one that runs.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/features/library/album_detail_screen.dart [hunk 5]
```diff
@@ -208,9 +190,10 @@ class _AlbumDetailScreenState extends ConsumerState<AlbumDetailScreen> {
       index: index,
       selectable: true,
       selectionActive: _selecting,
-      selected: _selectedUris.contains(track.uri),
+      selected: _selection.contains(track.uri),
       onSelectStart: () => _enterSelection(track),
       onSelectToggle: () => _toggle(track),
+      onSelectRange: _extendSelection,
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/features/library/album_detail_screen.dart:220]: Agreed — I added the modifier clicks to all three screens but the focus handling to only one. Fixed in 5a2596c.

Rather than copy it into two more screens, the handling is now `SelectionEscapeScope` in `library/widgets/`: it owns the node, claims focus on the rising edge of a selection, and stays out of the Tab order. All three screens use it, so `LibraryScreen` lost its bespoke `_selectionFocus` in the process.

New tests cover Escape on both detail pages, verified to fail without the scope.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/shared/widgets/context_menu_region.dart [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    if (items.isEmpty) return;
+    final RenderObject? overlay =
+        Overlay.of(context).context.findRenderObject();
+    if (overlay is! RenderBox) return;
+
+    final T? value = await showMenu<T>(
+      context: context,
+      position: RelativeRect.fromRect(
+        Rect.fromPoints(globalPosition, globalPosition),
+        Offset.zero & overlay.size,
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/shared/widgets/context_menu_region.dart]: Confirmed and fixed in 0e8f8ca — the anchor goes through `overlay.globalToLocal()` now.

Measured before fixing: a right-click 400 px in opened the menu at 612 px, a 212 px shift (the 200 px rail plus the menu's own padding). The existing menu tests missed it because they pump the library screen on its own, where the overlay sits at the origin and the offset is zero, so the new test pumps the shape the shell actually has — a fixed-width column, then a `Navigator` holding the row — and covers the pointer anchor, the keyboard one, and a click near the right edge.

The keyboard path shares `_openAt`, so converting there covers both; `_centreOf` stays global and is converted with everything else at the point of use.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/features/library/widgets/track_tile.dart [hunk 7]
```diff
@@ -264,109 +332,173 @@ class _OverflowMenu extends ConsumerWidget {
...
+  // something to it". Offered only where there is somewhere to go — a track
+  // with no album tags has no album page to open.
+  final bool hasAlbum = (track.albumName ?? '').trim().isNotEmpty ||
+      (track.albumId ?? '').trim().isNotEmpty;
+  final bool hasArtist = (track.artistName ?? '').trim().isNotEmpty;
+  if (hasAlbum || hasArtist) {
+    items.add(const PopupMenuDivider());
+    if (hasAlbum) {
+      items.add(
+          _item(_TrackAction.showAlbum, Icons.album_outlined, 'Show album'));
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/features/library/widgets/track_tile.dart:431]: Real, and reproduced. Fixed in b445710.

`FoldersScreen` builds its rows straight from `folderListingProvider`, which is an on-demand server listing with no relationship to the flat catalog, so a well-tagged track browsed before a sync got both actions and both landed on "not found".

The menu now gates them on catalog membership: two memoized id-set providers (`libraryAlbumIdsProvider`, `libraryArtistIdsProvider`) answer it per row without scanning every album on each open. The tag check stays, so an untagged track is still offered nowhere to go.

New test browses a folder over an empty catalog and asserts both actions are withheld while the track-level ones stay, then asserts they come back once the catalog holds the track. Verified it fails without the fix.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/features/player/mini_player.dart [hunk 8]
```diff
@@ -360,6 +378,22 @@ class _MiniProgressBar extends ConsumerWidget {
...
+      return PlaybackProgressBar(
+        // Same reasoning as the full player's bar: identity is the track, so a
+        // track change disposes an optimistically held seek instead of carrying
+        // it onto the next song.
+        key: ValueKey<String?>(state.currentTrack?.uri),
+        position: state.position,
+        duration: state.duration,
+        playing: state.isPlaying,
+        density: WavySeekBarDensity.compact,
+        showTimeLabels: false,
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/features/player/mini_player.dart:392]: Fixed in b445710, though the numbers are milder than the report says.

The slider path isn't at its default 48dp: the existing `SliderTheme` override already caps the overlay at radius 12, so it renders 24dp, not ~48. I flipped `defaultPlaybackProgressStyle` to `slider` locally and ran the player suite: nothing overflowed, and the four failures were pre-existing tests that look for `WavySeekBar` by type. So there was no broken layout to fix.

The gap that is real is the one in the API I added: `density` was documented as wave-only, which makes the escape hatch quietly worse in the one place that has no room. The slider now honours it too (track 2, thumb 3, overlay 6), taking the compact bar from 24dp to 12dp, just under the wave's 14.

Test pumps both densities side by side and asserts compact is smaller, is at most 16dp (64dp bar minus 48dp of transport), and still seeks. Both bars go in one tree deliberately: `_RenderSlider` only repaints on a theme change, so measuring one after the other in the same element reads the first size twice.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/features/player/widgets/queue_sheet.dart [hunk 9]
```diff
@@ -65,75 +75,77 @@ class QueueSheet extends ConsumerWidget {
...
+                    const SliverToBoxAdapter(
+                      child: SizedBox(height: AppSpacing.md),
+                    ),
+                  ],
+                ),
+        ),
+      ],
+    );
+
+    if (embedded) return body;
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/features/player/widgets/queue_sheet.dart:143]: Confirmed and fixed in 535ec09: the repository is read before the dialog now, next to the messenger that was already captured for exactly this reason.

Reproduced first. The new test opens the queue as a pane, starts the save, narrows the window past the pane minimum while the name prompt is up, then submits. Against the old code it throws `Bad state: Cannot use "ref" after the widget was disposed` and nothing is saved.

Worth noting why this only showed up now: as a bottom sheet the queue is a route, so it outlives its own dialogs. The pane made it an ordinary widget in the layout, and a resize takes it away mid-action.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/shared/layout/pane_layout.dart [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+  final double paneWidth;
+  final double minWidthForPane;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        final bool paneVisible = constraints.maxWidth >= minWidthForPane;
+        final Widget list = listBuilder(context, paneVisible);
+        if (!paneVisible) return list;
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/shared/layout/pane_layout.dart:84]: Right, and the doc comment right above that return claimed the opposite, which is a good sign it was wrong rather than deliberate. Fixed in 535ec09.

Went with lifting it: `LibraryScreen` now holds a `TrackSelection` per pane, beside the `_paneAlbumId` / `_paneArtistId` it already keeps for the same reason, and passes it down. Both detail screens take an optional `selection` and fall back to their own when pushed as a route, where the route's lifetime is the right one anyway. Picking a different album or artist still clears it, since a selection only means anything against the list it was made in.

Keeping the subtree mounted was the other option, but that pays for a live detail screen on every narrow window, and the pane is exactly where it is not wanted.

Three tests, verified to fail without it: the selection survives narrowing past `listDetailMinWidth` and back, it does not follow the pane onto another album, and the artists pane keeps its own separate from the albums one. The doc comment now says what the host has to do instead of claiming it comes for free.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/features/library/album_detail_screen.dart [hunk 11]
```diff
@@ -234,23 +241,23 @@ class _AlbumDetailScreenState extends ConsumerState<AlbumDetailScreen> {
...
-      if (!_selectedUris.add(track.uri)) {
-        _selectedUris.remove(track.uri);
-      }
+      _selection.toggle(track);
     });
   }
 
   void _exitSelection() {
-    setState(_selectedUris.clear);
+    setState(_selection.clear);
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/features/library/album_detail_screen.dart]: Correct, and it is a consequence of the previous commit: making the selection outlive the screen is exactly what turned that `mounted` guard from harmless into wrong. Fixed in 0003034.

`_exitSelection` now clears unconditionally and guards only the rebuild, on both detail screens. New test selects a track in the album pane, opens Add to playlist, narrows past `listDetailMinWidth` while the sheet is up, dismisses it, then widens: verified that against the old code the pane comes back showing "1 selected".

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/shared/widgets/context_menu_region.dart [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+    final T? value = await showMenu<T>(
+      context: context,
+      position: RelativeRect.fromRect(
+        Rect.fromPoints(anchor, anchor),
+        Offset.zero & overlay.size,
+      ),
+      items: items,
+    );
+    if (value != null) onSelected(value);
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply lib/shared/widgets/context_menu_region.dart]: Real. Fixed in 0003034: the dispatch is guarded on `context.mounted`.

Went with dropping the pick rather than capturing the dependencies. The action is a `T` handed to a caller-supplied callback, and what each caller reaches for differs (the row's ref, an ancestor for a dialog, a router), so there is nothing this widget could usefully capture on their behalf. A pick that lands after the surface is gone has no surface to act on.

Test pumps a region, opens the menu, unmounts the region while the popup is up, then picks: nothing dispatched, nothing thrown. Against the old code the callback fires into the dead tree. Second test covers the ordinary path, so the guard cannot quietly swallow a normal pick.

---
_Generated by [Claude Code](https://claude.ai/code)_
